test/filters/test_filters.cpp

```diff
@@ -2033,7 +2033,44 @@ TEST (FrustumCulling, Filters)
   removed = fc.getRemovedIndices ();
   EXPECT_EQ (removed->size (), input->size ());
 
+  // Should filter all points in the input cloud
+  fc.setNegative (false);
+  fc.setKeepOrganized (false);
+  fc.setRegionOfInterest (0.5f, 0.5f, 1.0f, 1.0f);
+  fc.filter (*output);
+  EXPECT_EQ (output->size (), input->size ());
+  removed = fc.getRemovedIndices ();
+  EXPECT_EQ (removed->size (), 0);
+  // Check invalid ROI values: no points should remain
+  fc.setRegionOfInterest (0.5f, 0.5f, 0.0f, 0.0f);
+  fc.filter (*output);
+  EXPECT_EQ (output->size (), 0);
+  removed = fc.getRemovedIndices ();
+  EXPECT_EQ (removed->size (), input->size ());
+  fc.setRegionOfInterest (-0.4f, 0.0f, 8.2f, -1.3f);
+  fc.filter (*output);
+  EXPECT_EQ (output->size (), 0);
+  removed = fc.getRemovedIndices ();
+  EXPECT_EQ (removed->size (), input->size ());
 
+  // Test on real point cloud, cut out milk cartoon in milk_cartoon_all_small_clorox.pcd
+  pcl::PointCloud <pcl::PointXYZ>::Ptr model (new pcl::PointCloud <pcl::PointXYZ>);
+  pcl::copyPointCloud (*cloud_organized, *model);
+
+  Eigen::Matrix4f cam2robot;
+  cam2robot << 0, 0, 1, 0, 0, -1, 0, 0, 1, 0, 0, 0, 0, 0, 0, 1;
+  // Cut out object based on ROI 
+  fc.setInputCloud (model);
+  fc.setNegative (false);
+  fc.setVerticalFOV (43);
+  fc.setHorizontalFOV (57);
+  fc.setNearPlaneDistance (0);
+  fc.setFarPlaneDistance (0.9);
+  fc.setRegionOfInterest (0.44f, 0.30f, 0.16f, 0.38f);
+  fc.setCameraPose (cam2robot);
+  fc.filter (*output);
+  // Should extract object; number of points based on milk.pcd
+  EXPECT_NEAR (output->size (), 13704, 13704 * 0.05); 
```

Comment:
This size shouldn't vary from time to time since its the same point cloud and there is no random involved here?
Is it required to have this "tolerance" value?